client/src/base/features/set-model.ts

```diff
@@ -21,7 +21,7 @@ export class RequestModelAction implements Action {
     static readonly KIND = 'requestModel'
     readonly kind = RequestModelAction.KIND
 
-    constructor(public readonly options?: { [key: string]: string }) {
+    constructor(public options?: { [key: string]: string }) {
```

Comment:
I'd love to keep at least the actions immutable. Can't we just create a new one below?

Reply:
We have a lot of actions that are _not_ immutable: InitializeCanvasBoundsAction, SetModelAction, SetBoundsAction, RequestBoundsAction, ComputedBoundsAction, SetPopupModelAction, UpdateModelAction, LoggingAction.

Such a `readonly` keyword is effective only when the action is created via constructor, but not when it's parsed from JSON.

Even if we make `options` read-only, its content are still mutable (the same is true for most other actions that hold objects or arrays).